react/src/tests/opReturn.test.ts

```diff
@@ -85,94 +107,109 @@ describe('getDataPushdata', () => {
     // (b(😂) = 4) +
     // (b(👌) = 4) +
     // = 13, 0d in hex
-    const data13bytes = 'a©æ😂👌'
-    expect(getDataPushdata(data13bytes)).toBe('0d')
-  })
+    const data13bytes = 'a©æ😂👌';
+    expect(getDataPushdata(data13bytes)).toBe('0d');
+  });
   it('Throws if data > 205 bytes', () => {
-    const data206bytes = '%'.repeat(206)
-    expect(() =>
-      getDataPushdata(data206bytes)
-    ).toThrow('Maximum 205 byte size exceeded for user data: 206')
-   })
-})
+    const data206bytes = '%'.repeat(206);
+    expect(() => getDataPushdata(data206bytes)).toThrow(
+      'Maximum 205 byte size exceeded for user data: 206',
+    );
+  });
+});
 
-describe('parseOpReturnProps', () => {
+describe('encodeOpReturnProps', () => {
   // protocol pushdata + protocol + version byte
-  const allResultsPrefix = '04' + '50415900' + '00'
-  const paymentIdDigits = 18
-  const paymentIdRegex = /^[0-9a-fA-F]{18}$/
-  it('Undefined data', () => {
-    const fullResult = parseOpReturnProps(undefined)
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix + '00') // 00 is pushdata for no data
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
+  const allResultsPrefix = '04' + '50415900' + '00';
+  const paymentIdDigits = 18;
+  const paymentIdRegex = /^[0-9a-fA-F]{18}$/;
+  it('Parses undefined data', () => {
+    const fullResult = encodeOpReturnProps(undefined);
+    const resultPaymentId = fullResult?.slice(-paymentIdDigits);
+    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits);
+    expect(resultNoPaymentId).toBe(allResultsPrefix + '00'); // 00 is pushdata for no data
```

Comment:
Shouldn't this be `allResultsPrefix + '00' + '00'` now, per the latest spec? There is no data and no nonce. But the fields are still required.

Reply:
No, this is the slice of the part that only contains the data. The slice that contains the paymentId contains its pushdata; two lines above. I've changed the variable names to make it more clear.